src/components/SignupForm/index.js

```diff
@@ -1,79 +1,93 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useForm } from 'react-hook-form';
+import { t } from 'i18next';
 
 import logo from '../../app/assets/logo.png';
 import ButtonForm from '../ButtonForm';
 import InputText from '../InputText';
 import { isPasswordEqual, emailPattern } from '../../utils/formValidations';
+import { signUp } from '../../services/userService';
+import useRequest from '../../app/hooks/useRequest';
+import { AUTH_INPUTS } from '../../constants/forms';
 
-import { AUTH_INPUTS } from './fields';
 import styles from './styles.module.scss';
 
 function SignupForm() {
+  const [userData, setUserData] = useState();
   const { register, handleSubmit, errors, getValues } = useForm();
-
+  const [isLoading, submitError] = useRequest({ request: signUp, payload: userData });
   const onSubmit = handleSubmit(({ firstName, lastName, email, password, confirmPassword }) => {
-    const user = {
+    setUserData({
       firstName,
       lastName,
       email,
       password,
       confirmPassword
-    };
-    console.log({ user });
+    });
```

Comment:
when do you fire the request?

Reply:
the request is fired when we change the value of payload. If is necesary call again we can use sendRequest(payload)

Reply:
Your useRequest hook should be able to give the component responsibility for when it wants to execute it